sklearn/utils/tests/test_validation.py

```diff
@@ -387,12 +387,15 @@ def test_check_array_dtype_warning():
         assert_equal(X_checked.dtype, np.float64)
 
     for X in float64_data:
-        X_checked = assert_no_warnings(check_array, X, dtype=np.float64,
-                                       accept_sparse=True, warn_on_dtype=True)
-        assert_equal(X_checked.dtype, np.float64)
-        X_checked = assert_no_warnings(check_array, X, dtype=np.float64,
-                                       accept_sparse=True, warn_on_dtype=False)
-        assert_equal(X_checked.dtype, np.float64)
+        with pytest.warns(DeprecationWarning,
```

Comment:
For asserting that no warnings are raises you should:

- ignore the DeprecationWarning like before
- use `with pytest.warn(None)`

Reply:
done